### ISSUE TITLE
Terragrunt 0.67.0 errors out with specific messages logged to stderr

### ISSUE THREAD

**Issue by camlow325** [gruntwork-io/terragrunt#3372]
## Describe the bug

Terragrunt `0.67.0` errors out when messages with specific formats are logged to `stderr`.

## Steps To Reproduce

Add the following files to the current directory:

**terragrunt.hcl**
```hcl
terraform {
  before_hook "test" {
    commands = ["apply"]
    execute  = ["./test.sh"]
  }
}
```
**test.sh**
```bash
#!/usr/bin/env bash

echo "foo [bar] baz" >&2
```

**test.tf**
```terraform
# empty
```

1. Run `terragrunt apply`.

## Expected behavior

Terraform / OpenTofu apply succeeds, with this message in the output:

```
No changes. Your infrastructure matches the configuration.
```

## Actual Behavior

Apply errors out with this message:

```
09:15:53.451 ERROR  Error running hook test with message: [...] parsing time "foo" as "2006-01-02T15:04:05.000-0700": cannot parse "foo" as "2006"
09:15:53.451 ERROR  Errors encountered running before_hooks. Not running 'terraform'.
09:15:53.758 ERROR  1 error occurred:
        * [...] parsing time "foo" as "2006-01-02T15:04:05.000-0700": cannot parse "foo" as "2006"
```

As a slight variation, add the following lines to the end of the `test.sh` script:

```
sleep 5
echo "done sleeping" >&2
```

With these changes in place, a `terragrunt apply` results in the following error instead:

```
09:26:42.371 ERROR  Error running hook test with message: [...] signal: broken pipe
09:26:42.371 ERROR  Errors encountered running before_hooks. Not running 'terraform'.
09:26:42.707 ERROR  1 error occurred:
        * [...] signal: broken pipe
```

In cases where the script which writes the message that triggers the error is called from a higher-level script, we've also seen this error message:

```
09:30:49.002 ERROR  Error running hook setup with message: [...] exit status 141
09:30:49.002 ERROR  Errors encountered running before_hooks. Not running 'terraform'.
09:30:49.734 ERROR  1 error occurred:
        * [...] exit status 141
```

## Versions

- Terragrunt version: `0.67.0`
- OpenTofu/Terraform version: `OpenTofu 1.8.1`, `Terraform 1.9.5`
- Environment details (Ubuntu 20.04, Windows 10, etc.): `macOS Sonoma 14.6.1`

## Additional context

This problem does not occur on earlier versions of Terragrunt, including `v0.66.9`. I'm guessing that the problem is due to changes made in #3338. It seems to occur when messages which match [this regex](https://github.com/gruntwork-io/terragrunt/blob/v0.67.0/internal/log/tf_writer.go#L25) are written to `stderr`.

While we've only experienced this problem so far in the context of scripts that write messages to `stderr` from hooks, it seems like this problem could occur in other contexts, e.g., custom error messages generated during an OpenTofu/Terraform failure. Where we first saw this error, the bracket syntax was used to output a resource address containing an index, e.g., `module.foo[0]`.

We've experimented with adding the `--terragrunt-forward-tf-stdout` and/or  `--terragrunt-disable-log-formatting` commands to the Terragrunt command line, but the error persisted when a message matching the regex is written to `stderr`.

**Comment by yhakbar** [comment]
Hey @camlow325 ,

Thanks for reporting this. We're taking a look at this, and will look to get a fix in soon.

**Comment by dehanjl** [comment]
Can confirm that I am getting the same issue. Had to downgrade to Terragrunt `v0.66.9`.

**Comment by levkohimins** [comment]
Resolved in [v0.67.5](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.67.5) release.
